From f9018cbd7deb34fa33791981232d024f6f144e47 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:05 -0400
Subject: [scsi] qla2xxx: Use the correct mailbox registers when acknowledging an IDC request on ISP8044

Message-id: <1408391817-6389-18-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87918
O-Subject: [PATCH RHEL 7.1 e-stor 17/69] qla2xxx: Use the correct mailbox registers when acknowledging an IDC request on ISP8044.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

scsi commit id 9e5054ec6b6296075b757d0ea0d3e9e25784db20

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_isr.c b/drivers/scsi/qla2xxx/qla_isr.c
index 2e6eae3..41de008 100644
--- a/drivers/scsi/qla2xxx/qla_isr.c
+++ b/drivers/scsi/qla2xxx/qla_isr.c
@@ -269,11 +269,18 @@ qla81xx_idc_event(scsi_qla_host_t *vha, uint16_t aen, uint16_t descr)
   { "Complete", "Request Notification", "Time Extension" };
  int rval;
  struct device_reg_24xx __iomem *reg24 = &vha->hw->iobase->isp24;
+ struct device_reg_82xx __iomem *reg82 = &vha->hw->iobase->isp82;
  uint16_t __iomem *wptr;
  uint16_t cnt, timeout, mb[QLA_IDC_ACK_REGS];
 
  /* Seed data -- mailbox1 -> mailbox7. */
- wptr = (uint16_t __iomem *)&reg24->mailbox1;
+ if (IS_QLA81XX(vha->hw) || IS_QLA83XX(vha->hw))
+  wptr = (uint16_t __iomem *)&reg24->mailbox1;
+ else if (IS_QLA8044(vha->hw))
+  wptr = (uint16_t __iomem *)&reg82->mailbox_out[1];
+ else
+  return;
+
  for (cnt = 0; cnt < QLA_IDC_ACK_REGS; cnt++, wptr++)
   mb[cnt] = RD_REG_WORD(wptr);
 
-- 
1.7.1